wordpress-mobile/WordPress-iOS
# Reader: Add new header style for the tag stream page

Refs pcdRpT-3Eb-p2
Depends on #21475

This partially implements the new header style for the Tag Stream page. I tried using the SwiftUI but it's not going to work without a decent refactor in the `ReaderStreamViewController+Helper` because it expects the header to be a `UIView` that conforms to `ReaderStreamHeader`.

Additionally, I tried implementing the horizontal layout by following a simple rule: we'll use the horizontal layout if the tag title is less than 10 characters AND the user doesn't use any accessibility categories. Let me know how it works, I can revert it and always use vertical layout if it's clunky. cc: @osullivanchris 

Here are some previews:

### Before

<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/28991e7e-2dcf-42ed-b6c3-90f38d9dcd59" width=350>

### After

Short titles | Longer titles
-|-
![Simulator Screenshot - iPhone 14 Pro - 2023-08-31 at 18 30 00](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/952ebc16-44a4-4c06-a48e-db2375c77684) | ![Simulator Screenshot - iPhone 14 Pro - 2023-08-31 at 18 30 47](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/5bca8121-d0b6-4ca1-ae2f-d944b22ea3c1)

⚠️ A couple notes ⚠️ 

- The Follow/Unfollow button is still using the old design, as per the breakdown suggested in p1693342310830629/1693330435.413469-slack-C05N140C8H5.
- The metadata information is not included yet in this PR, as it requires an additional API endpoint.
- I noticed that the header has a bottom border on the design. However, I decided to exclude this for now because we are still using the "old" layout for the overall stream page; there is a gap between the header and each card. Let me know if you'd prefer the border to be added!

## To test

- Launch the Jetpack app.
- Make sure the `Reader Improvements v1` feature flag is turned on.
- Go to Reader > Discover, and select any recommended tag.
- 🔎 Verify that the new header design is shown.
- 🔎 Verify that the follow/unfollow functionality works.
- Go back and try finding a tag with longer text. Usually, you can find these displayed on the top section of any reader card.
- 🔎 Verify that header section is displayed using vertical layout.

## Regression Notes
1. Potential unintended areas of impact
Should be none. The feature is hidden behind a feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


UI Changes testing checklist:
